src/extensibility/ExtensionManagerView.js

```diff
@@ -337,6 +338,25 @@ define(function (require, exports, module) {
             context[helper] = registry_utils[helper];
         });
 
+        // Do some extra validation on homepage url to make sure we don't end up executing local binary
+        if (context.metadata.homepage) {
+            var parsed = PathUtils.parseUrl(context.metadata.homepage);
+
+            // We can't rely on path-utils because of known problems with protocol identification
+            // Falling back to Browsers protocol identification mechanism
+            var tmpLink = document.createElement('a');
```

Comment:
Do you want to create an 'a' element everytime. You can resuse it? The GC of V8 is memory hungry, so will never know when this will be GC'ed.